### DISCUSSION TITLE
Haiku OS Terminal Port

### DISCUSSION THREAD

**Discussion by jeremypass96** [catppuccin/catppuccin#2161] [Port Requests]
### Will you make the port?

- [x] Yes, I'm willing to make the port.

### How can the suggested port be themed?

Info about Haiku Terminal: https://www.haiku-os.org/docs/userguide/en/applications/terminal.html

Where Haiku Terminal themes go in the OS: /boot/system/settings/Terminal/Themes

Example theme file: https://github.com/haiku/haiku/blob/master/data/system/data/terminal_themes/Default

This would *probably* need to be made into a Haiku .hpkg to install it on the Haiku OS, though probably not. Just the theme files themselves should probably work fine.

### Submission Guidelines

- [X] I have read and followed Catppuccin's submission guidelines.

**Comment by jeremypass96** [comment]
[Catppuccin.Frappé.txt](https://github.com/catppuccin/catppuccin/files/12602009/Catppuccin.Frappe.txt)
[Catppuccin Latte.txt](https://github.com/catppuccin/catppuccin/files/12591265/Catppuccin.Latte.txt)
[Catppuccin Macchiato.txt](https://github.com/catppuccin/catppuccin/files/12591266/Catppuccin.Macchiato.txt)
[Catppuccin.Mocha.txt](https://github.com/catppuccin/catppuccin/files/12601994/Catppuccin.Mocha.txt)


Here are the ported themes. Not sure if they're 100% correct, as I am unable to test them. Putting them in /boot/system/settings/Terminal/Themes does nothing (for some reason). They don't show up in the "Color scheme" pop-up menu dialog under settings.

**Reply by humdingerb** [reply]
Works for me (Haiku hrev57257 nightly build).
Your "Frappé" has some encoding problem with the theme name's "é". Should be UTF8 encoded, or it's "Catppuccin Frapp" in Terminal's Colors menu.

**Comment by humdingerb** [comment]
Oh, and the "Mocha" has the wrong theme name - also says the encoding challenged "_Catppuccin Frappé_".

**Reply by jeremypass96** [reply]
I've fixed it.

**Comment by Begasus** [comment]
This could work, could you check @humdingerb (I'm on R1B4, so it doesn't work there):
```
SUMMARY="Soothing pastel theme for the high-spirited"
DESCRIPTION="Catppuccin is a community-driven pastel theme that aims to be the middle ground \
between low and high contrast themes. It consists of 4 soothing warm flavors with 26 eye-candy \
colors each, perfect for coding, designing, and much more! In addition, this repository tracks \
the development of the actual color palette, the project's documentation, organization-wide \
assets, resources and code samples for maintainers/developers."
HOMEPAGE="https://homepage/of/projectx.org"
COPYRIGHT="2021 Catppuccin"
LICENSE="MIT"
REVISION="1"
SOURCE_URI="https://github.com/catppuccin/catppuccin/files/12591264/Catppuccin.Frappe.txt#noarchive"
CHECKSUM_SHA256="f556320ff02decfe5029d41d70e61fba01e3847f2a3a87362ca3d50bad632627"

SOURCE_URI_2="https://github.com/catppuccin/catppuccin/files/12591265/Catppuccin.Latte.txt#noarchive"
CHECKSUM_SHA256_2="9001d39daef839f5bcd005380d2ed69919b145dfbbdabbaaf42ee275eef6e4de"

SOURCE_URI_3="https://github.com/catppuccin/catppuccin/files/12591266/Catppuccin.Macchiato.txt#noarchive"
CHECKSUM_SHA256_3="219ee84708febdef000203cd9f0ce65332ce9bba739af1e786c8389abb80dd0a"

SOURCE_URI_4="https://github.com/catppuccin/catppuccin/files/12591267/Catppuccin.Mocha.txt#noarchive"
CHECKSUM_SHA256_4="7b1495baa533fa8b7735ed4d6a8131130a5393679d88b7a7cb6f9d0233f24e4d"
SOURCE_DIR=""

ARCHITECTURES="all"

GLOBAL_WRITABLE_FILES="
	settings/Terminal/Themes directory keep-old
	"

PROVIDES="
	catppuccin = $portVersion
	"

INSTALL()
{
	mkdir -p $settingsDir/Terminal/Themes
	cp Catppuccin.Frappe.txt $settingsDir/Terminal/Themes
	cp $sourceDir2/Catppuccin.Latte.txt $settingsDir/Terminal/Themes
	cp $sourceDir3/Catppuccin.Macchiato.txt $settingsDir/Terminal/Themes
	cp $sourceDir4/Catppuccin.Mocha.txt $settingsDir/Terminal/Themes
}
```

The files are grabbed in the comment above, so I don't think that would be a final source link?

**Reply by humdingerb** [reply]
When it comes to creating the real recipe, the copied files should be renamed to lose the ".txt".

**Reply by Begasus** [reply]
That's only a small change, the re-naming in the 2 files are needed though (as you mentioned earlier), otherwise only 3 will show up in the menu.

**Reply by jeremypass96** [reply]
Update the link for Mocha: https://github.com/catppuccin/catppuccin/files/12601994/Catppuccin.Mocha.txt
And the link for Frappé: https://github.com/catppuccin/catppuccin/files/12602009/Catppuccin.Frappe.txt

**Reply by jeremypass96** [reply]
Homepage should be: https://github.com/catppuccin/

**Reply by Begasus** [reply]
Thanks, as mentioned earlier, this was a test recipe, nothing has been  done  at haikuports yet, if things are good it can be pushed and the package will be available for install. :)

**Comment by Begasus** [comment]
Works :)

![catppuccin](https://github.com/catppuccin/catppuccin/assets/16057090/1291af5d-8afc-4c5f-a125-581a8306ee0a)

**Reply by jeremypass96** [reply]
Frappé has been fixed.

**Comment by Begasus** [comment]
Completed in https://github.com/haikuports/haikuports/pull/9416 :)

**Comment by jeremypass96** [comment]
Colors were wrong. I've updated and fixed them.

[Catppuccin Frappé.txt](https://github.com/catppuccin/catppuccin/files/13687072/Catppuccin.Frappe.txt)
[Catppuccin Latte.txt](https://github.com/catppuccin/catppuccin/files/13687073/Catppuccin.Latte.txt)
[Catppuccin Macchiato.txt](https://github.com/catppuccin/catppuccin/files/13687074/Catppuccin.Macchiato.txt)
[Catppuccin Mocha.txt](https://github.com/catppuccin/catppuccin/files/13687075/Catppuccin.Mocha.txt)

**Reply by Begasus** [reply]
Done in https://github.com/haikuports/haikuports/pull/9892

**Comment by jeremypass96** [comment]
**Oops!** They weren't _quite_ correct. They are **100% correct** now!

[Catppuccin Frappé.txt](https://github.com/catppuccin/catppuccin/files/13690471/Catppuccin.Frappe.txt)
[Catppuccin Latte.txt](https://github.com/catppuccin/catppuccin/files/13690472/Catppuccin.Latte.txt)
[Catppuccin Mocha.txt](https://github.com/catppuccin/catppuccin/files/13690473/Catppuccin.Mocha.txt)
[Catppuccin Macchiato.txt](https://github.com/catppuccin/catppuccin/files/13690474/Catppuccin.Macchiato.txt)

**Reply by Begasus** [reply]
Sure these are the ones that need to be used, just pushed and it seems those links are different then the ones showing now?

**Reply by jeremypass96** [reply]
Yes.

**Reply by Begasus** [reply]
Could you double check to see if these are correct? https://github.com/haikuports/haikuports/pull/9896

**Reply by jeremypass96** [reply]
Yes, these are correct.

**Reply by Begasus** [reply]
Thanks, merged now :)
